### ISSUE TITLE
DQN not learning on Cartpole

### ISSUE THREAD

**Issue by txzhao** [reinforceio/tensorforce#264]
Hi,

First of all, thanks for open-sourcing this repository. It's really an amazing work! I'm actually trying the example code quickstart.py and modify it a little bit to train a DQN agent on CartPole v0 using the parameters below, and it seems that the agent is not learning anything at all (average reward ~20). Could someone share a suitable set of parameters for this case? Or is there something wrong with my parameter settings? Thanks in advance.

Cheers,
Tianxiao

The parameters I used is listed below:
![capture](https://user-images.githubusercontent.com/23320220/33831428-64ce0168-de78-11e7-80b0-60673310055e.PNG)

**Comment by michaelschaarschmidt** [comment]
Hello,

in general, please do not create issues asking about configurations unless you have reason to believe there is a bug (read the FAQ for why).  

Something like this should work:

{
    "type": "dqn_agent",
    "batch_size": 32,
    "batched_observe": 1,

    "memory": {
        "type": "replay",
        "capacity": 50000,
        "random_sampling": true
    },

    "optimizer": {
        "type": "clipped_step",
        "clipping_value": 0.01,
        "optimizer": {
            "type": "adam",
            "learning_rate": 1e-3
        }
    },

    "discount": 1.0,

    "variable_noise": null,

    "first_update": 1000,
    "update_frequency": 1,
    "repeat_update": 1,

    "huber_loss": 1.0,

    "target_sync_frequency": 500,
    "target_update_weight": 1.0,

    "double_q_model": true,

    "explorations_spec": {
        "type": "epsilon_anneal",
        "initial_epsilon": 1.00,
        "final_epsilon": 0.02,
        "timesteps": 1e4
    },
}

 "network": [
        {
            "type": "dense",
            "size": 64,
            "activation": "relu"
        }
    ]

[2017-12-11 15:03:45,315] Finished episode 500 after 200 timesteps. Steps Per Second 221.089668269
[2017-12-11 15:03:45,315] Episode reward: 200.0
[2017-12-11 15:03:45,315] Average of last 500 rewards: 41.234
[2017-12-11 15:03:45,316] Average of last 100 rewards: 80.45


Finally, consider contribution benchmark results to the benchmark repo, where performance is best discussed.

**Comment by AlexKuhnle** [comment]
[Gitter](https://docs.google.com/forms/d/1_UD5Pb5LaPVUviD0pO0fFcEnx_vwenvuc00jmP2rRIc/viewform?edit_requested=true) is a good place to ask about configuration issues.

**Comment by txzhao** [comment]
Thanks! I'll give it a try. And so sorry for creating issue here! I'll keep that in mind, and next time use Gitter instead.
